### PR TITLE
{Release} Upgrade to Azure CLI 2.12.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

File: src/azure-cli-core/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+2.12.0
+++++++
+* Add a new global parameter query-examples (#14765)
```
Comment (jiasli) [inline src/azure-cli-core/HISTORY.rst:8]:
```suggestion
* Add a new global parameter `--query-examples` (#14765)
```

**Review by jiasli — APPROVED** [review]

**Review by haroldrandom — APPROVED** [review]
Network LGTM

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,125 @@
...
+
+**App Config**
+
+* Support enabling/disabling PublicNetworkAccess for existing stores (#14966)
+
+**App Service**
+
+* Add support for Premium V3 pricing tier (#15055)
+* Fix #12653: az webapp log config --application-logging false doesn't turn it off (#14929)
+* : fix #14684- access-restriction remove by ip address does not work; #13837-az webapp create - Example for different RSgroups for Plan and WebApp (#14818)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:34]:
```suggestion
Fix #14684: access-restriction remove by ip address does not work (#14818)
Fix #13837: az webapp create - Example for different RSgroups for Plan and WebApp (#14818)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,125 @@
...
+
+**ARM**
+
+* `az deployment group/sub/mg/tenant what-if`: Show "Ignore" resource changes last (#14932)
+
+**Compute**
+
+* Add new license_type in vm create/update: RHEL_BYOS, SLES_BYOS (#14934)
+* Upgrade disk API version to 2020-06-30 and some new features (#15031)
+* vm boot-diagnostics enable: --storage becomes optional; New command: vm boot-diagnostics get-boot-log-uris (#15145)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:46]:
```suggestion
* vm boot-diagnostics enable: --storage becomes optional (#15145)
* New command: vm boot-diagnostics get-boot-log-uris (#15145)
* vm boot-diagnostics get-boot-log: support managed storage (#15145)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,125 @@
...
+
+* Support enabling/disabling PublicNetworkAccess for existing stores (#14966)
+
+**App Service**
+
+* Add support for Premium V3 pricing tier (#15055)
+* Fix #12653: az webapp log config --application-logging false doesn't turn it off (#14929)
+* : fix #14684- access-restriction remove by ip address does not work; #13837-az webapp create - Example for different RSgroups for Plan and WebApp (#14818)
+* functionapp: Added support for custom handlers. Deprecated Powershell 6.2. (#15170)
+* functionapp: Fixed issue where app setting was being incorrectly set for linux custom images (#15153)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:36]:
```suggestion
* functionapp: Add support for custom handlers. Deprecated Powershell 6.2. (#15170)
* functionapp: Fix issue where app setting was being incorrectly set for linux custom images (#15153)
```

File: src/azure-cli/HISTORY.rst (hunk 2 above)
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:34]:
```suggestion
* Fix #14684- access-restriction remove by ip address does not work; #13837-az webapp create - Example for different RSgroups for Plan and WebApp (#14818)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,125 @@
...
+
+**Config**
+
+* Rename local-context to config param-persist (#15068)
+
+**Cosmos DB**
+
+* Support for Migration APIs for Throughput resource for Autoscale feature in CosmosDB (#14887)
+
+**Extension**
```
Comment (Juliehzl) [inline src/azure-cli/HISTORY.rst:56]:
```suggestion
**Eventhub**

Added Cluster commands and trusted_service_access_enabled parameter for Networkruleset

**Extension**
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,125 @@
...
+
+* `az extension add`: Add --upgrade option to update the extension if already installed (#15083)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:58]:
```suggestion
* `az extension add`: Add `--upgrade` option to update the extension if already installed (#15083)
* Turn on dynamic install by default (#15081)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,125 @@
...
+* functionapp: Fixed issue where app setting was being incorrectly set for linux custom images (#15153)
+
+**ARM**
+
+* `az deployment group/sub/mg/tenant what-if`: Show "Ignore" resource changes last (#14932)
+
+**Compute**
+
+* Add new license_type in vm create/update: RHEL_BYOS, SLES_BYOS (#14934)
+* Upgrade disk API version to 2020-06-30 and some new features (#15031)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:45]:
```suggestion
* Upgrade disk API version to 2020-06-30 (#15031)
* disk create: add --logical-sector-size, --tier (#15031)
* disk update: Support --disk-iops-read-only, --disk-mbps-read-only, --max-shares (#15031)
* new command disk-encryption-set list-associated-resources (#15031)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,130 @@
...
+
+**App Config**
+
+* Support enabling/disabling PublicNetworkAccess for existing stores (#14966)
+
+**App Service**
+
+* Add support for Premium V3 pricing tier (#15055)
+* Fix #12653: az webapp log config --application-logging false doesn't turn it off (#14929)
+* Fix #14684- access-restriction remove by ip address does not work; #13837-az webapp create - Example for different RSgroups for Plan and WebApp (#14818)
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst:34]:
```suggestion
* Fix #14684: access-restriction remove by ip address does not work; #13837-az webapp create - Example for different RSgroups for Plan and WebApp (#14818)
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,125 @@
...
+
+* `az extension add`: Add --upgrade option to update the extension if already installed (#15083)
+
+**IoT**
+
+* Enabled minimum TLS version on IoT Hub Create (#14882)
+
+**IoT Central**
+
+* CLI: App delete operation should be long running operation (#15161)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:71]:
```suggestion
* App delete operation is now long running operation (#15161)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,135 @@
...
+
+* Fix #14811 Add support for dockerignore override (#14916)
+
+**AKS**
+
+* CLI should tolerate empty kubeconfig (#14914)
+* FIX #12871: az aks enable-addons: Autogenerated help example is wrong for vitual-node option (#14989)
+* Remove legacy aci connector actions (#15084)
+* Support azure policy addon in azure-cli (#15092)
+* Fixed case sensitive issue for AKS dashboard addon (#15123)
```
Comment (fengzhou-msft) [inline src/azure-cli/HISTORY.rst:19]:
```suggestion
* Fix case sensitive issue for AKS dashboard addon (#15123)
```

**Review by fengzhou-msft — APPROVED** [review]

**Review by arrownj — APPROVED** [review]

**Review by qwordy — APPROVED** [review]

**Review by houk-ms — APPROVED** [review]

**Comment by yonzhan** [comment]
Release

**Pull request merged**